package-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultAssessmentParserIT.java

```
@@ -0,0 +1,77 @@
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opentestsystem.rdw.ingest.common.test.CachingTest;
+import org.opentestsystem.rdw.ingest.processor.PackageProcessorApplication;
+import org.opentestsystem.rdw.ingest.processor.model.Assessment;
+import org.opentestsystem.rdw.ingest.processor.service.DataElementErrorCollector;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@Transactional
+@CachingTest
+@ContextConfiguration(classes = {PackageProcessorApplication.class})
+public class DefaultAssessmentParserIT {
+
+    @Autowired
+    private DefaultAssessmentParser assessmentParser;
+
+
+    // naming convention of itShould<full sentence in camel case>()
+    @Test
+    public void itShouldCatchAllErrorsAndReturnNoObjectsWhenParsingBadData() {
+        DataElementErrorCollector errorCollector = new DataElementErrorCollector();
+        HashSet<Assessment> asmtSet = new HashSet<Assessment>();
+        assertThat(errorCollector.isEmpty());
+        assertThat(asmtSet.isEmpty());
+        try {
+            final Iterable<CSVRecord> records =
+                    CSVFormat.RFC4180.withFirstRecordAsHeader().parse(new InputStreamReader(this.getClass().getResourceAsStream("/IAB_English-bad.items.csv")));
+
+            for (CSVRecord record : records) {
+                asmtSet.add(assessmentParser.parse(record, errorCollector));
+            }
+        }catch(IOException ioe) {
+
```

if i see empty catch block in a test, i usually stick in a `fail(ioe.getMessage())`